scripts/search_index.py

```diff
@@ -47,10 +36,14 @@ def index_flexdown(source: str, href: str) -> list[tuple[str, str, str]]:
 
     # Get the blocks in the source code.
     blocks = flexdown.get_blocks(source)
-    flexdown.process_blocks(blocks, env)
 
     # Get all the markdown blocks.
-    markdown_blocks = [block for block in blocks if block.type == MarkdownBlock.type]
+    markdown_blocks = [block for block in blocks if isinstance(block, MarkdownBlock)]
+
+    # Do not do any transforms - it is only possible during render (which we don't do here).
```

Comment:
Ah I think maybe we need to `render` the blocks, otherwise the template expressions get left in.

Maybe we can do something similar to the code

```python
# The environment used for execing and evaling code.
env: types.Env = source.metadata

# Get the content of the document.
source = source.content

# Render each block.
markdown_blocks = []
for block in self.get_blocks(source, filename):
    try:
        out.append(block.render(env=env))
        if isinstance(block, MarkdownBlock):
            markdown_blocks.append(block)
    except Exception as e:
        print(
            f"Error while rendering {type(block)} on line {block.start_line_number}. "
            f"\n{block.get_content(env)}"
        )
        raise e
 ```